native-sql-engine/core/src/main/scala/com/intel/oap/expression/ColumnarUnaryOperator.scala

```diff
@@ -541,6 +555,15 @@ class ColumnarCast(
             TreeBuilder.makeFunction("castDECIMAL", Lists.newArrayList(child_node), dType)
           (funcNode, dType)
       }
+    } else if (dataType == TimestampType) {
```

Comment:
Current code already has the support to cast LongType, DateType, StringType to TimeStampType.
It looks the cast should be tackled according to child type in this piece of code.